components/eam/src/chemistry/modal_aero/modal_aero_amicphys.F90

```diff
@@ -4481,6 +4475,8 @@ subroutine modal_aero_amicphys_init( imozart, species_class,n_so4_monolayers_pca
     nspec_amode, &
     numptr_amode, numptrcw_amode, sigmag_amode
 
+use modal_aero_coag, only: set_coagulation_pairs
```

Comment:
There is another comment below about `use modal_aero_amicphys_control` in `modal_aero_coag.F90`. I've addressed that one by changing to `use ... only: ...`.

This comments, in contrast, is about `modal_aero_amicphys.F90`. Since `modal_aero_amicphys_control.F90` contains pretty much all the module variables originally declared in `modal_aero_amicphys.F90`, I think in a `use ... only: ...` statement we will end up having to list everything (i.e., 100+ variables :-) ). I would suggest we take the second option mentioned above by @singhbalwinder, i.e., defer this to a future PR. Also note that in the box model, we have split those 100+ module variables into groups related to the aerosol modes, species, individual processes, and the mapping of tracer indices between `amicphys` and the host model. It might be useful to discuss (for a future PR) whether we want to port those changes back to the global model.

Reply:
Let's defer that to the interface PR. For the rename process, I also saw that some variables (e.g. log of `sigmag`) are redefined in amicphys for all the modes. There may be other similar calculations that we can directly use from modal_aero_data.F90 or other places in the model. This will help us bring down the number of variables in amicphys_control